core/src/main/java/hudson/model/PeriodicWork.java

```diff
@@ -99,15 +102,49 @@ public long getInitialDelay() {
     @Initializer(after= JOB_LOADED)
     public static void init() {
         // start all PeriodicWorks
-        for (PeriodicWork p : PeriodicWork.all()) {
-            Timer.get().scheduleAtFixedRate(p, p.getInitialDelay(), p.getRecurrencePeriod(), TimeUnit.MILLISECONDS);
+        ExtensionList<PeriodicWork> extensionList = all();
+        extensionList.addListener(new PeriodicWorkExtensionListListener(extensionList));
+        for (PeriodicWork p : extensionList) {
+            schedulePeriodicWork(p);
         }
     }
 
+    private static void schedulePeriodicWork(PeriodicWork p) {
+        Timer.get().scheduleAtFixedRate(p, p.getInitialDelay(), p.getRecurrencePeriod(), TimeUnit.MILLISECONDS);
+    }
+
     // time constants
     protected static final long MIN = 1000*60;
     protected static final long HOUR =60*MIN;
     protected static final long DAY = 24*HOUR;
 
     private static final Random RANDOM = new Random();
+
+    /**
+     * ExtensionListener that will kick off any new AperiodWork extensions from plugins that are dynamically
+     * loaded.
+     */
+    private static class PeriodicWorkExtensionListListener extends ExtensionListListener {
+
+        private final Set<PeriodicWork> registered = new HashSet<>();
+
+        PeriodicWorkExtensionListListener(ExtensionList<PeriodicWork> initiallyRegistered) {
+            for (PeriodicWork p : initiallyRegistered) {
+                registered.add(p);
+            }
+        }
+
+        @Override
+        public void onChange() {
+            synchronized (registered) {
+                for (PeriodicWork p : PeriodicWork.all()) {
+                    // it is possibly to programatically remove Extensions but that is rarely used.
```

Comment:
this at least improves the current situation for dynamically loaded plugins.  Any plugins being programatically removed would still be scheduled (which is no different to what would happen without this change)